src/syrupy/report.py

```diff
@@ -98,23 +143,25 @@ def unused(self) -> "SnapshotFossils":
             self.discovered, self.used
         ):
             snapshot_location = unused_snapshot_fossil.location
-            if self.is_providing_paths and not any(
-                TestLocation(node).matches_snapshot_location(snapshot_location)
-                for node in self.ran_items
+            if self._provided_test_paths and not self._selected_items_match_location(
+                snapshot_location
             ):
+                # Paths/Packages were provided to pytest and the snapshot location
+                # does not match therefore ignore this unused snapshot fossil file
                 continue
 
-            if self.ran_all_collected_tests:
+            provided_nodes = self._get_matching_path_nodes(snapshot_location)
+            if self.ran_all_collected_tests and not any(provided_nodes):
+                # All collected tests were run and files were not filtered by ::node
+                # therefore the snapshot fossil file at this location can be deleted
                 unused_snapshots = {*unused_snapshot_fossil}
                 mark_for_removal = snapshot_location not in self.used
             else:
                 unused_snapshots = {
                     snapshot
                     for snapshot in unused_snapshot_fossil
-                    if any(
-                        TestLocation(node).matches_snapshot_name(snapshot.name)
-                        for node in self.ran_items
-                    )
+                    if self._selected_items_match_name(snapshot.name)
+                    or self._provided_nodes_match_name(snapshot.name, provided_nodes)
```

Comment:
This is the main gist of the fix. Basically `_selected_items_match_name` use the keyword expressions when they are present or matches against the ran items when no keywords provided or `_provided_nodes_match_name` checks if the nodes given for the test file matching this fossil location matches the snapshot name.